xml/nvset-4.0.rng

```diff
@@ -44,12 +44,7 @@
       <zeroOrMore>
         <element name="nvpair">
           <choice>
-            <group>
-              <attribute name="id-ref"><data type="IDREF"/></attribute>
-              <optional>
-                <attribute name="name"><text/></attribute>
-              </optional>
-            </group>
+            <attribute name="id-ref"><data type="IDREF"/></attribute>
```

Comment:
@gao-yan , "This feature was added by commit https://github.com/ClusterLabs/pacemaker/commit/3912538a8954158d44a57f23aa219c5529a170ab but is undocumented and
inconsistently implemented." Do you think this feature needs to be kept? A transform can preserve the effect of existing values

Reply:
Thanks for the heads-up, @kgaillot .

> @gao-yan , "This feature was added by commit [3912538](https://github.com/ClusterLabs/pacemaker/commit/3912538a8954158d44a57f23aa219c5529a170ab) but is undocumented and inconsistently implemented." 

"Attribute reference with a different name specified" is only documented in the crmsh man page:
https://crmsh.github.io/man-4.6/#topics_Features_AttributeReferences

, although it is rather a demonstration of the idea than a practical example, since "apache" RA doesn't actually support a "server_ip" parameter. 


> Do you think this feature needs to be kept?

I don't remember whether I implemented this because of any real demands or only for potential needs :-) I was asking people here if they were aware of any actual use cases in practice. No one has answered yes so far.

 > A transform can preserve the effect of existing values

You mean via XSLT upon upgrade of "validate-with"? Sounds sensible.

Reply:
Yep